src/device/device_ledger.cpp

```diff
@@ -1145,19 +1144,12 @@ namespace hw {
         hw::ledger::check32("derivation_to_scalar", "res", res_x.data, res_clear.data);
         #endif
 
-        unlock_device();
-      }catch (...)  {
-        unlock_device();
-        throw;
-      }
-      return true;
+        return true;
     }
 
     bool device_ledger::derive_secret_key(const crypto::key_derivation &derivation, const std::size_t output_index, const crypto::secret_key &sec, crypto::secret_key &derived_sec) {
-      lock_device();
```

Comment:
Doesn't this function also require the same branch logic with has_view_key?

Reply:
No (if I correctly analyse the code). Once the TX identified, the whole process is done throw device, even the re-computation of derivation encrypted from device